wg-api/best-practices.md

```diff
@@ -134,6 +141,26 @@ Even if seconds (or some other time unit) are more natural in the domain of an A
 
 See https://w3ctag.github.io/design-principles/#milliseconds
 
+### Provide create/read/update/delete options when dealing with arrays
+
+If an API is designed to accept an array of items, consider providing methods of creating, reading, updating, and deleting items.
+
+In the case of third-party libraries, one might want to add a single item rather than replacing existings items.
+
+```js
+// Bad: third-party libraries can only replace registered schemes
+protocol.registerSchemesAsPrivileged([
+  { scheme: 'app', privileges: { standard: true } }
+])
+
+// Good: third-party libraries can create, read, update, and delete items
+if (protocol.getRegisteredSchemes().includes(scheme => scheme.scheme === 'app')) {
+  protocol.unregisterScheme({ scheme: 'app' })
+}
+protocol.registerScheme({ scheme: 'app', privileges: { standard: true } })
+protocol.updateScheme({ scheme: 'app', privileges: { secure: true } })
```

Comment:
I know this API doesn't actually exist, but if we were to implement this API, would there be a reason to have `unregisterScheme` take a dictionary rather than just being `protocol.unregisterScheme('app')`? I can't think of any other properties we'd ever want to add to this dictionary.

Also suggesting tweaks to API names since we're potentially going to point that PR at these docs as guidance for their API. There's an important distinction between registering schemes (can happen at any time) and registering schemes as privileged (must happen before `ready` event).

```suggestion
// Good: third-party libraries can create, read, update, and delete items
if (protocol.getPrivilegedSchemes().includes(scheme => scheme.scheme === 'app')) {
  protocol.unregisterSchemeAsPrivileged('app' )
}
protocol.registerSchemeAsPrivileged({ scheme: 'app', privileges: { standard: true } })
protocol.updatePrivilegedScheme({ scheme: 'app', privileges: { secure: true } })
```